spokestack/mic/pyaudio.py

```diff
@@ -0,0 +1,51 @@
+"""
+This module uses pyaudio to receive input from a
+device microphone
+"""
+import pyaudio  # type: ignore
+
+from spokestack.config import SpeechConfig
+
+
+class PyAudioMicrophoneInput:
+    """
+    Retrieve microphone input with PyAudio
+
+    Args:
+        config (SpeechConfig): configuration paramaters
+    """
+
+    def __init__(self, config: SpeechConfig) -> None:
+
+        self.config = config
```

Comment:
for a library like this, it would be better to make these attributes private (_ prefix)
for properties that you want to expose to the client, make the attribute private and create an @property method that provides read-only access